### ISSUE TITLE
"Seeds" and "Peers" columns are the other way around and slightly confusing

### ISSUE THREAD

**Issue by Kanjirito** [johman10/flood-for-transmission#336]
Currently "Seeds" is `SEEDING_TO / PEERS_CONNECTED` but it should be `DOWNLOADING_FROM / PEERS_CONNECTED`. A "Seeds" value of 5/10 should mean that there are 10 available clients seeding the torrent and we are downloading from 5 of them. That's the way all other UIs (that I have seen) do it. "Peers" has the opposite problem.

The other thing is that `PEERS_CONNECTED` doesn't _really_ tell you how many clients are downloading or seeding. It's just the amount of clients we are currently connected to. If there are only 10 clients downloading and none seeding, for some users it might be confusing why there are apparently 10 seeders but nothing is being downloaded ("Seeds" and "Peers" would show 0/10). One solution would be to use `leecherCount` and `seederCount` which are in [`trackerStats`](https://github.com/transmission/transmission/blob/main/extras/rpc-spec.txt#L362-L367). `trackerStats` returns an array for every tracker but it should be safe to assume that each tracker from a torrent has the same amount of leechers and seeders connected to it. This solution could cause a different confusion though. If someone has closed ports they might see clients seeding but won't be able to download from them. Personally I think it makes more sense to use the leecher and seeder count but it's pretty subjective.
Another solution could be just removing `PEERS_CONNECTED` from both "Seeds" and "Peers" and just add it as a separate column.

**Comment by Kanjirito** [comment]
I can definitely see why you did it that way. I wouldn't say it's wrong but I think that being more inline with other torrent UIs will be easier to understand.  "Seeding to x of y" is more verbose than "Seeds x of y" so there's no way to really misread it. The way it is worded now to me reads as "Connected to x seeders of y". I guess the column name could be changed to "Downloading from".
The `PEERS_CONNECTED` part is not as simple though. It's a very valid information to show but as I said, it does not really tell us what type of connection we have and it is capped by the max connections setting (which defaults to 50). As it is now you can't know how many clients are actually using that torrent and if anyone is seeding it. The first problem generally isn't an issue but maybe someone doesn't want to seed a torrent that has 2k seeders already and would rather seed something else. A "Total peer count" column would solve that problem, maybe also a "Leechers/Seeders" column.

If I was making the choice I would make the columns like this:
- "Seeds" (or "Seeders")  = `peersSendingToUs / seederCount`
- "Peers" = `peersGettingFromUs / leecherCount`
- Connected Peers = `peersConnected`

**Comment by johman10** [comment]
Hmm... You're making good points!

Perhaps I would vouch for renaming the current columns though so I would go for:
"Peers" => "Downloading from"
"Seeds" => "Seeding to"

And then I can add extra columns called:
"Total leechers"
"Total seeders"
These will take their information from the tracker to show the data you're asking for. This naming convention would also be more in line with the Transmission API and so potentially take away a bit of the confusion.

Would that clarify it well enough for you?

**Comment by johman10** [comment]
Looking into this a little further, it also seems like the values for leecherCount and seederCount aren't super trustworthy. I see quite a lot of cases where the value is `-1` while the download actually works.

**Comment by Kanjirito** [comment]
That would definitely make it clearer, yes. It makes sense to try and be more in line with the default web UI. I'm fine with that.
The leecher and seeder count is a bit behind when starting since it has to be retrieved from the tracker but in practice it's not really an issue. It does not need to be super accurate all the time. I'm also not sure if that's something all trackers support but if it's separate columns you can just make the value `0` or `N/A` (the latter would be clearer) if it returns `-1`. Instead of making it 2 columns it could also be a single `leecherCount / seederCount` column but those numbers can be 4 digits each so it's probably a bad idea. There could also be a seeders / leechers ratio column but that seems mostly pointless.

**Comment by johman10** [comment]
I vouched for two seperate columns for Total Leechers and Total Seeders. They are based on the maximum amount found on one of the trackerStats. If all of them are `-1` it will show up as `-`.

I also renamed the columns Peers and Seeds to Downloading from and Seeding to. I hope this clarifies it better, and accomodates a good experience!

Thanks for your input, and let me know if you have more thoughts/concerns!
